source/common/http/utility.cc

```diff
@@ -1047,11 +1057,15 @@ std::string Utility::localPathFromFilePath(const absl::string_view& file_path) {
   return absl::StrCat("/", file_path);
 }
 
-RequestMessagePtr Utility::prepareHeaders(const envoy::config::core::v3::HttpUri& http_uri) {
-  absl::string_view host, path;
-  extractHostPathFromUri(http_uri.uri(), host, path);
+RequestMessagePtr Utility::prepareHeaders(const envoy::config::core::v3::HttpUri& http_uri,
+                                          bool include_scheme) {
+  absl::string_view scheme, host, path;
+  extractSchemeHostPathFromUri(http_uri.uri(), scheme, host, path);
 
   RequestMessagePtr message(new RequestMessageImpl());
+  if (include_scheme) {
```

Comment:
Should this also check that the scheme isn't empty?
I guess that if it is empty, the header will be added with an empty value. However, the difference will be the existence of that key in the header-map.